### PR TITLE
feat: insert products as silver labels

### PR DESCRIPTION
**Pull request by alejandrods**
This PR inserts the list of products as silver labels in the candidate generation step. In addition, the `prepare_dataset.py` script has been improved to filter out reviews that are in another language and to remove html tags. Also remove the different patterns identified for each review_source. It also includes some logs to debug the training process.

This needs another PR to improve model training and clean up the repository.

### CODE REVIEW

File: src/preprocess/annotator_base.py [hunk 1]
```diff
@@ -18,15 +18,23 @@ def __init__(self, preprocessor: Preprocessor, use_cache=True):
...
         sents = marked_doc['sents']
+        # Here, we should insert the negative keyphrases such as `product solving`
         for sent in sents:
             phrases = sent['phrases']
             assert phrases
             positive_spans = [tuple(phrase[0]) for phrase in phrases]
             num_positive = len(positive_spans)
             # sample negatives
             word_idxs = sent['widxs']
+            # Iterave over all ngrams of length MINGRAMS to MAXGRAMS
```
Comment (dilekkisa) [inline src/preprocess/annotator_base.py]: typo: Iterate

File: src/preprocess/annotator_base.py [hunk 2]
```diff
@@ -18,15 +18,23 @@ def __init__(self, preprocessor: Preprocessor, use_cache=True):
...
+            # (0, 4), (1, 5)...
+            # (0, 3), (1, 4), ...
+            # (0, 2), (1, 3), ...
+            # (0, 1), (1, 2), ...
             all_spans = utils.get_possible_spans(word_idxs, len(sent['ids']), consts.MAX_WORD_GRAM, consts.MAX_SUBWORD_GRAM)
+            # Discard positive spans in negative_spans
```
Comment (dilekkisa) [inline src/preprocess/annotator_base.py:37]: does this mean, if there is a positive sample in a negative sample, we discard the positive sample?

**Review by dilekkisa — APPROVED** [review]
Great work on the comments to understand and decode the implementation! 🎖️ 

Dropped a single question inline.

File: src/preprocess/annotator_core.py [hunk 3]
```diff
@@ -10,25 +11,38 @@
...
 
 MINCOUNT = 2
-MINGRAMS = 1
+MINGRAMS = 2
 MAXGRAMS = consts.MAX_WORD_GRAM
 
 
 PUNCS_SET = set(string.punctuation) - {'-'}
 STPWD_SET = set(utils.TextFile.readlines('../data/stopwords.txt'))
+with open('../data/products_found_in_staging.pkl', 'rb') as f:
```
Comment (dilekkisa) [inline src/preprocess/annotator_core.py:20]: Is the code for creating this file here?

Comment (alejandrods) [reply src/preprocess/annotator_core.py:20]: No, it is for reading the list of products which is generated in the notebook. This is why I have created an [issue](https://github.com/kluein/klue/issues/11021) to re-organize/clean/create a new repository to simplify the training process.

File: src/preprocess/annotator_base.py (hunk 2 above)
Comment (alejandrods) [reply src/preprocess/annotator_base.py:37]: No, this part simply generates the negative samples. So the steps are:
- Create all possible spans in a variable called `all_spans`
- Remove the **positive spans** from the `all_spans` to not consider positive spans as negative.
- Select randomly negative spans from `all_spans`

File: src/preprocess/annotator_base.py (hunk 1 above)
Comment (alejandrods) [reply src/preprocess/annotator_base.py]: Fixed it! Thanks

**Pull request merged**